### PR TITLE
transcribe hcal test beam pedestals

### LINKED ISSUE #1: Transcribe HCal Test Beam Conditions
Need to extract conditions from elog and data files floating around.

### LINKED ISSUE #5: Fill in Missing Values for Pedestal
Need to add in missing channels to all of the pedestal files and either input expected values, or very very large values to effectively turn off the channel in analysis

### LINKED ISSUE #7: 2 Incorrect Pedestal Files.
Files `pedestals_20220416.csv` and `pedestals_20220422.csv` both have the incorrect pedestal data inside them. For `pedestals_20220416.csv`, the first half of the channels use the 04-15 pedestal data instead of 04-16. For `pedestals_20220422.csv`, it looks like it uses 04-21 instead of 04-22 for the first half.

### PR DESCRIPTION
**Pull request by tomeichlersmith**
Just opening this draft PR for a location to chat about these files. @cjfroemming @jmmans

### CODE REVIEW

File: Hcal/testbeam04-2022/pedestals/hcal_detids_hex.txt [hunk 1]
```diff
@@ -0,0 +1,385 @@
+DetID
```
Comment (tomeichlersmith) [inline Hcal/testbeam04-2022/pedestals/hcal_detids_hex.txt]: This file may be interesting to keep around. We could use it to double check any tables being added/changed in the future.

File: Hcal/testbeam04-2022/pedestals/index_v1_0_0.csv [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+125,137,"ANY","Hcal/testbeam04-2022/pedestals/v1_0_0/pedestals_20220415.csv"
+138,163,"ANY","Hcal/testbeam04-2022/pedestals/v1_0_0/pedestals_20220416.csv"
+164,204,"ANY","Hcal/testbeam04-2022/pedestals/v1_0_0/pedestals_20220417.csv"
+205,224,"ANY","Hcal/testbeam04-2022/pedestals/v1_0_0/pedestals_20220419.csv"
+224,234,"ANY","Hcal/testbeam04-2022/pedestals/v1_0_0/pedestals_20220421.csv"
+235,257,"ANY","Hcal/testbeam04-2022/pedestals/v1_0_0/pedestals_20220422.csv"
+258,272,"ANY","Hcal/testbeam04-2022/pedestals/v1_0_0/pedestals_20220423.csv"
+273,281,"ANY","Hcal/testbeam04-2022/pedestals/v1_0_0/pedestals_20220424_07.csv"
+282,298,"ANY","Hcal/testbeam04-2022/pedestals/v1_0_0/pedestals_20220424_22.csv"
+299,315,"ANY","Hcal/testbeam04-2022/pedestals/v1_0_0/pedestals_20220426.csv"
```
Comment (tomeichlersmith) [inline Hcal/testbeam04-2022/pedestals/index_v1_0_0.csv:13]: Need to either update these paths or move the data files themselves.

File: Hcal/testbeam04-2022/pedestals/testbeam22_missing_channel_freq.csv [hunk 3]
```diff
@@ -0,0 +1,123 @@
+,PEDESTAL_ADC,PEDESTAL_RMS_ADC
```
Comment (tomeichlersmith) [inline Hcal/testbeam04-2022/pedestals/testbeam22_missing_channel_freq.csv]: Is this the frequency of missing or pedestals for the missing channels?

File: Hcal/testbeam04-2022/pedestals/testbeam22_missing_pedestals.ipynb [hunk 4]
```diff
@@ -0,0 +1,272 @@
+{
```
Comment (tomeichlersmith) [inline Hcal/testbeam04-2022/pedestals/testbeam22_missing_pedestals.ipynb]: Thank you for sharing this, I will take a look, but we will probably drop this before merging to keep the git repo clean and focused on just the data tables.

File: Hcal/testbeam04-2022/pedestals/testbeam22_pedestals.csv [hunk 5]
```diff
@@ -0,0 +1,12 @@
+# Hand-constructed for testing purposes May 3 2022
```
Comment (tomeichlersmith) [inline Hcal/testbeam04-2022/pedestals/testbeam22_pedestals.csv]: Will remove this in favor of the other index file.

File: Hcal/testbeam04-2022/pedestals/testbeam22_run_info.csv [hunk 6]
```diff
@@ -0,0 +1,58 @@
+# Made by hand May 16th, 2022
+# Time is in ISO format (YYYY-MM-DDTHH:MM)
```
Comment (tomeichlersmith) [inline Hcal/testbeam04-2022/pedestals/testbeam22_run_info.csv]: Move this to its own directory (maybe `Beam`) to separate it from the Hcal subsystem.

File: Hcal/testbeam04-2022/pedestals/testbeam22_run_ts_events.csv [hunk 7]
```diff
@@ -0,0 +1,77 @@
+# Made by hand May 16th, 2022
+# TS_EVENTS are in thousands
```
Comment (tomeichlersmith) [inline Hcal/testbeam04-2022/pedestals/testbeam22_run_ts_events.csv]: Not sure what to do with this file.

**Review by tomeichlersmith — COMMENTED** [review]
Great additions Cooper! I have many comments but don't feel like any one of us need to address all of them. We can all weigh in here and make easy changes when we get a chance.

File: Hcal/testbeam04-2022/pedestals/testbeam22_run_ts_events.csv (hunk 7 above)
Comment (cjfroemming) [reply Hcal/testbeam04-2022/pedestals/testbeam22_run_ts_events.csv]: Joe asked for a file with TS events separately. We could probably do without it at some point, however. I may just send it to him directly.

**Comment by tomeichlersmith** [comment]
Did a good amount of reorganization. All that is left is to resolve issue #7 which seems to just require a re-run of the original pedestal analyzer and then a re-construction of the full pedestal table using Cooper's jupyter notebook.

**Comment by tomeichlersmith** [comment]
Ok, I've rerun the extractor script which worked for 04/16 pedestals. 04/22 pedestals are more complicated because, unfortunately, the two raw pedestal files for that day are bit-level identical. I think this is because one of them got accidentally copied over when moving files around, but in any case, **04/22 doesn't have any pedestals for DPM 1** as a first pass, I just copied the 04/21 pedestals for DPM 1.

I think we could do a more detailed study to attempt to "reconstruct" DPM1's pedestal for 04/22, but that can be easily fixed by introducing a future version of the pedestal conditions (e.g. v1_1_0) which uses v1_0_0 for all pedestals except 04/22.

**Pull request merged**